back/boxtribute_server/business_logic/statistics/crud.py

```diff
@@ -91,16 +126,83 @@ def compute_created_boxes(base_id=None):
         if row["created_on"] is not None:
             row["created_on"] = row["created_on"].date()
 
-    products_ids = {f["product_id"] for f in facts}
-    dimensions = {
-        "product": list(
-            Product.select(Product.id, Product.name)
-            .where(Product.id << products_ids)
-            .dicts()
-        ),
-        "category": list(
-            ProductCategory.select(ProductCategory.id, ProductCategory.name).dicts()
-        ),
-    }
+    dimensions = _generate_dimensions("category", "product", facts=facts)
+    return {"facts": facts, "dimensions": dimensions}
+
+
+def compute_top_products_checked_out(base_id):
+    """Return list of most-checked-out products (i.e. highest count in transactions)
+    with rank included, grouped by distribution date and product category.
+    """
+    selection = Transaction.select(
+        Transaction.created_on.alias("checked_out_on"),
+        Transaction.product.alias("product_id"),
+        Product.category.alias("category_id"),
+        fn.SUM(Transaction.count).alias("items_count"),
+    ).join(
+        Product, on=((Product.base == base_id) & (Transaction.product == Product.id))
+    )
+    facts = (
+        selection.group_by(SQL("product_id"), SQL("category_id"), SQL("checked_out_on"))
+        .order_by(SQL("items_count").desc())
+        .dicts()
+    )
+
+    # Data transformations
+    for rank, row in enumerate(facts, start=1):
+        row["rank"] = rank
+        row["checked_out_on"] = row["checked_out_on"].date()
+
+    dimensions = _generate_dimensions("category", "product", facts=facts)
+    dimensions["size"] = None
+    return {"facts": facts, "dimensions": dimensions}
+
+
+def compute_top_products_donated(base_id):
+    """Return list of most-donated products with rank included, grouped by distribution
```

Comment:
Will we also count boxes assigned to shipments to it in the future?

Reply:
User story in #903 does not mention this, maybe additional thoughts from @aerinsol ?

Reply:
i put a comment about this into #903